src/write-batch.js

```diff
@@ -266,7 +266,7 @@ class WriteBatch {
 
     validate.isDocumentReference('documentRef', documentRef);
     validate.isDocument('data', data, {
-      allowEmpty: !merge,
+      allowEmpty: true,
```

Comment:
Is this allowEmpty member even required anymore?

Reply:
It's still used for update() calls.